### ISSUE TITLE
Search or Replace Field Should Not Be Scrollable by Gestures with Single-Line Content

### ISSUE THREAD

**Issue by honood** [zed-industries/zed#15622]
### Check for existing issues

- [X] Completed

#### Steps to Reproduce

1.  Open any existing file or create a new file using Zed.app.
2.  Open the search field (the issue is the same with the replace field).
3.  Do not enter any characters in the search field (the placeholder will be displayed), or enter any characters.
4.  Scroll within the search field using the mouse or trackpad gestures.

- **Expected Result:** The placeholder or characters in the input field should not scroll upwards with the gesture.
- **Actual Result:** The placeholder or characters in the input field can scroll out of the visible area with the gesture.

### Environment

Zed: v0.146.3 (Zed)
OS: macOS 14.5.0
Memory: 64 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

https://github.com/user-attachments/assets/2da26e98-8992-404b-b49a-203408f968fa

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by notpeter** [comment]
Interesting.  I am not able to reproduce this on my Macbook with the internal trackpad.

Are you by chance using an external mouse/trackpad or any software which might customize your scroll events?

**Comment by honood** [comment]
@notpeter Yes, I use Apple Magic Mouse (connected via Bluetooth), but I don’t use any third-party software to configure the mouse or trackpad.

**Comment by notpeter** [comment]
I believe this may have been fixed in the intervening months, I was not able to reproduce the behavior scrolling with my Apple Trackpad, Apple Magicmouse or Logitech Trackball scrollwheel.

Thanks for reporting!
